File: web/src/api/catalog.ts
import type {
  SupportedExchangePayload,
  SupportedProviderTypePayload,
} from "@/types/public"
import request from "@/utils/request"

const Api = {
  SupportedProviderTypes: "/api/catalog/supported-provider-types",
  SupportedExchanges: "/api/catalog/supported-exchanges",
} as const

export function getSupportedProviderTypesApi() {
  return request.get<SupportedProviderTypePayload[]>(Api.SupportedProviderTypes)
}

export function getSupportedExchangesApi() {
  return request.get<SupportedExchangePayload[]>(Api.SupportedExchanges)
}

File: core/src/routes/catalog.rs
use axum::{Router, routing::get};

use crate::{http::handlers::catalog::*, state::AppState};

pub fn router() -> Router<AppState> {
    Router::new()
        .route("/supported-provider-types", get(supported_provider_types))
        .route("/supported-exchanges", get(supported_exchanges))
}
